src/main/java/org/broadinstitute/consent/http/service/EmailService.java

```diff
@@ -233,6 +241,71 @@ private void sendNewDARRequestEmail(
+    EmailType emailType = EmailType.DAR_EXPIRED;
+    String interval = "1 year";
+    String noEmailForUserToWarnFoundLogTemplate = "User %d (%s) not found for expiring DAR.  Reference id: %s";
+    // Per value in ticket DT-1573
+    Timestamp minimumSubmittedDateForExpirations = Timestamp.from(Instant.ofEpochSecond(
+        LocalDate.of(2024, 9, 30).toEpochSecond(LocalTime.of(0, 0, 0, 0), ZoneOffset.UTC)));
+    List<DataAccessRequest> expiredDars = dataAccessRequestDAO.findAgedDARsByEmailTypeOlderThanInterval(
+        emailType.getTypeInt(), interval, minimumSubmittedDateForExpirations);
+    expiredDars.forEach(expiredDar -> {
+      try {
+        String referenceId = expiredDar.getReferenceId();
+        User user = userDAO.findUserById(expiredDar.getUserId());
+        String darCode = expiredDar.getDarCode();
+        String userName = user.getDisplayName();
+        if (user.getEmail() == null) {
+          // Do not throw here.  Log information about the DAR since this will continue
+          // to appear broken until manual intervention is taken to resolve the missing user
+          // email address
+          logWarn(String.format(noEmailForUserToWarnFoundLogTemplate,
+              expiredDar.getUserId(), userName, referenceId));
+        } else {
+          sendDarExpiredMessage(user, darCode, user.getUserId(), referenceId);
+        }
+      } catch (Exception e) {
+        logException(e);
+      }
+    });
+  }
+
+  private void sendDARExpirationReminderNotices() {
+    EmailType emailType = EmailType.DAR_EXPIRATION_REMINDER;
+    String interval = "11 months";
+    String noEmailForUserToWarnFoundLogTemplate = "User %d (%s) not found for expiring warning.  DAR reference id: %s";
+    // Per value in ticket DT-1573
+    Timestamp minimumSubmittedDateForExpirations = Timestamp.from(Instant.ofEpochSecond(
+        LocalDate.of(2024, 9, 30).toEpochSecond(LocalTime.of(0, 0, 0, 0), ZoneOffset.UTC)));
+    List<DataAccessRequest> expiredDars = dataAccessRequestDAO.findAgedDARsByEmailTypeOlderThanInterval(
+        emailType.getTypeInt(), interval, minimumSubmittedDateForExpirations);
+    expiredDars.forEach(expiredDar -> {
+      try {
+        String referenceId = expiredDar.getReferenceId();
+        User user = userDAO.findUserById(expiredDar.getUserId());
+        String darCode = expiredDar.getDarCode();
+        String userName = user.getDisplayName();
+        if (user.getEmail() == null) {
+          // Do not throw here.  Log information about the DAR since this will continue
+          // to appear broken until manual intervention is taken to resolve the missing user
+          // email address
+          logWarn(String.format(noEmailForUserToWarnFoundLogTemplate,
+              expiredDar.getUserId(), userName, referenceId));
+        } else {
+          sendDarExpirationReminderMessage(user, darCode, user.getUserId(), referenceId);
+        }
+      } catch (Exception e) {
+        logException(e);
```

Comment:
This code logs; do we expect to do anything with these logs? Since the API request is blocking, we could collect the errors, send them back in the response and log them in the cron service, where they might be more likely to be seen. The response could also include the number of emails sent of each type, etc.

Reply:
I think we're going to need to do things with our logs, regardless of what is causing entries in them or which logger created them.
I like the idea of logging the metrics.  I think the program could benefit more from doing that overall.

Reply:
The `logException()` code path triggers a sentry event so we'll see all errors in slack.